### PR TITLE
feat(fuse): add system$clustering_information function

### LINKED ISSUE #5439: clustering_information(<database>, <table>)
**Summary**

Returns clustering information, including average clustering depth.

```
call system$clustering_information(<database>, <table>)
```

The syntax `call system$clustering_information(<database>, <table>, <cluster_keys>)` will be implemented in the next pr.

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

```
mysql> create TABLE t(A int,  B int ) cluster by(B,A);
Query OK, 0 rows affected (0.05 sec)
Read 0 rows, 0.00 B in 0.006 sec., 0 rows/sec., 0.00 B/sec.

mysql> insert into t values(0,3),(1,1);
Query OK, 0 rows affected (0.03 sec)
Read 2 rows, 16.00 B in 0.011 sec., 175.4 rows/sec., 1.37 KiB/sec.

mysql> insert into t values(1,3),(2,1);
Query OK, 0 rows affected (0.06 sec)
Read 2 rows, 16.00 B in 0.012 sec., 165.46 rows/sec., 1.29 KiB/sec.

mysql> insert into t values(4,4) ;
Query OK, 0 rows affected (0.05 sec)
Read 1 rows, 8.00 B in 0.012 sec., 82.9 rows/sec., 663.20 B/sec.

mysql> call system$clustering_information('default','t')\G
*************************** 1. row ***************************
           cluster_by_keys: (B, A)
         total_block_count: 3
total_constant_block_count: 1
          average_overlaps: 0.6667
             average_depth: 1.6667
     block_depth_histogram: {"00001":1,"00002":2}
1 row in set (0.07 sec)
Read 0 rows, 0.00 B in 0.013 sec., 0 rows/sec., 0.00 B/sec.
```

## Changelog

- New Feature

## Related Issues

Fixes #5439

### CODE REVIEW

File: query/src/storages/fuse/table_functions/clustering_informations/clustering_information.rs [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            Series::from_data(vec![info.block_depth_histogram]),
+        ]))
+    }
+
+    fn get_min_max_stats(&self, block: &BlockMeta) -> Result<(Vec<DataValue>, Vec<DataValue>)> {
+        if self.table.cluster_keys() != self.cluster_keys || block.cluster_stats.is_none() {
+            todo!()
+        }
+
+        let cluster_stats = block.cluster_stats.clone().unwrap();
```
[the remark this answers is not in the document]
Comment (zhyass) [reply query/src/storages/fuse/table_functions/clustering_informations/clustering_information.rs:98]: checked here 
https://github.com/datafuselabs/databend/blob/a0f5686c119131d1a06bb3107397aa4d5836721a/query/src/storages/fuse/table_functions/clustering_informations/clustering_information.rs#L93

File: tests/suites/0_stateless/09_fuse_engine/09_0014_func_clustering_information_function.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+-- Create table t09_0014
+create table t09_0014(a int, b int) cluster by(b,a);
+
+insert into t09_0014 values(0,3),(1,1);
+insert into t09_0014 values(1,3),(2,1);
+insert into t09_0014 values(4,4);
+
+select *  from t09_0014 order by b, a;
+
+--Bug in cluster mode: https://github.com/datafuselabs/databend/issues/5473
```
Comment (BohuTANG) [inline tests/suites/0_stateless/09_fuse_engine/09_0014_func_clustering_information_function.sql:10]: We can skip the test in cluster mode with a '_cluster' suffix new test file, let's run only in the stand-alone mode.

File: tests/suites/0_stateless/09_fuse_engine/09_0014_func_clustering_information_function_cluster.result [hunk 3]
```diff
@@ -0,0 +1,6 @@
+1	1
+2	1
+0	3
+1	3
+4	4
+ERROR 1105 (HY000) at line 9: Code: 1002, displayText = Unimplement error.
```
Comment (BohuTANG) [inline tests/suites/0_stateless/09_fuse_engine/09_0014_func_clustering_information_function_cluster.result:6]: This error message is not stable with the `at line 9:`.
Can we set an ErrCode number for it?  @zhang2014 @sundy-li

Comment (BohuTANG) [inline tests/suites/0_stateless/09_fuse_engine/09_0014_func_clustering_information_function_cluster.result:6]: This error message is not stable with the `at line 9:`.
Can we set an ErrCode number for it?  @zhang2014 @sundy-li

Comment (sundy-li) [reply tests/suites/0_stateless/09_fuse_engine/09_0014_func_clustering_information_function_cluster.result:6]: Why not use ` -- {ErrorCode 1002}`

Comment (BohuTANG) [reply tests/suites/0_stateless/09_fuse_engine/09_0014_func_clustering_information_function_cluster.result:6]: The `-- {ErrorCode 1002}` should be set on the `09_0014_func_clustering_information_function.sql`, but this test should not work with standalone.

Comment (zhyass) [reply tests/suites/0_stateless/09_fuse_engine/09_0014_func_clustering_information_function_cluster.result:6]: Let me comment out the case first, this bug will be fixed soon.

**Comment by BohuTANG** [comment]
Let's merge first.

**Pull request merged**